src/screens/SendSynthetic/SendSyntheticAmount.js

```diff
@@ -17,236 +17,190 @@
-`;
-
-const ActionsWrapper = styled.View`
-  display: flex;
-  flex-direction: row;
-  justify-content: space-between;
-`;
-
-const HelperText = styled(BaseText)`
-  ${fontStyles.medium};
-  color: ${themedColors.secondaryText};
-  margin-left: 4px;
-`;
-
-const TextRow = styled.View`
-  flex-direction: row;
-`;
-
-const ImageIcon = styled(CachedImage)`
-  width: 6px;
-  height: 12px;
-  tint-color: ${themedColors.primary};
-`;
 
 class SendSyntheticAmount extends React.Component<Props, State> {
-  syntheticsForm: t.form;
-  receiver: string;
-  receiverEnsName: string;
   source: string;
-  assetData: AssetData;
-  availableSyntheticBalance: number;
-  syntheticExchangeRate: number;
 
   constructor(props: Props) {
     super(props);
     const {
       navigation: { getParam: getNavigationParam },
-      availableSyntheticAssets,
     } = props;
 
     this.source = getNavigationParam('source', '');
-    this.receiver = getNavigationParam('receiver', '');
-    this.receiverEnsName = getNavigationParam('receiverEnsName');
-
-    this.assetData = getNavigationParam('assetData', {});
-    const fetchedSyntheticAsset = availableSyntheticAssets.find(({ symbol }) => symbol === this.assetData.token);
-    this.availableSyntheticBalance = get(fetchedSyntheticAsset, 'availableBalance', 0);
-    this.syntheticExchangeRate = get(fetchedSyntheticAsset, 'exchangeRate', 0);
-
-    const intentError = !this.availableSyntheticBalance
-      ? 'Asset has no available liquidity'
-      : null;
 
     this.state = {
-      intentError,
+      intentError: '',
       submitPressed: false,
       value: null,
       inputHasError: false,
+      selectedContact: null,
+      assetData: null,
+      receiver: '',
+      receiverEnsName: '',
     };
 
-    this.handleFormChange = debounce(this.handleFormChange, 500);
+    this.handleAssetValueSelect = debounce(this.handleAssetValueSelect, 500);
   }
 
   componentDidMount() {
-    this.props.initSyntheticsService();
-    this.props.fetchSingleAssetRates(this.assetData.token);
+    const { initSyntheticsService, fetchAvailableSyntheticAssets } = this.props;
+    initSyntheticsService();
+    fetchAvailableSyntheticAssets();
+    this.setPreselectedValues();
   }
 
+  setPreselectedValues = () => {
+    const { navigation, contactsByWallet } = this.props;
+    const contact = navigation.getParam('contact');
+    if (contact) {
+      const formattedContact = contactsByWallet.find(({ name }) => name === contact.username);
+      if (formattedContact) this.setReceiver(formattedContact);
+    }
+  };
+
+  handleReceiverSelect = (value: Option, onSuccess?: () => void) => {
+    const { navigation } = this.props;
+    const { name, hasSmartWallet } = value;
+
+    if (hasSmartWallet) {
+      this.setReceiver(value, onSuccess);
+    } else {
+      Alert.alert(
+        'This user is not on Pillar Network',
+        'You both should be connected to Pillar Network in order to be able to send instant transactions for free',
```

Comment:
Remember to suggest what the user could do: "Why not ask this user to install the Pillar Wallet?" for example...